ledger/src/blockstore.rs

```diff
@@ -1416,32 +1416,23 @@ impl Blockstore {
 
         if !erasure_meta.check_coding_shred(&shred) {
             metrics.num_coding_shreds_invalid_erasure_config += 1;
-            let conflicting_shred = self.find_conflicting_coding_shred(
-                &shred,
-                slot,
-                erasure_meta,
-                just_received_shreds,
-            );
-            if let Some(conflicting_shred) = conflicting_shred {
-                if !self.has_duplicate_shreds_in_slot(slot) {
-                    if self
-                        .store_duplicate_slot(
-                            slot,
-                            conflicting_shred.clone(),
-                            shred.payload().clone(),
-                        )
-                        .is_err()
-                    {
-                        warn!("bad duplicate store..");
-                    }
+            if !self.has_duplicate_shreds_in_slot(slot) {
+                let conflicting_shred = self
+                    .find_conflicting_coding_shred(&shred, slot, erasure_meta, just_received_shreds)
+                    .expect("Shred indicated by erasure meta must exist")
+                    .into_owned();
 
-                    duplicate_shreds.push(PossibleDuplicateShred::ErasureConflict(
-                        shred.clone(),
-                        conflicting_shred,
-                    ));
+                if self
+                    .store_duplicate_slot(slot, conflicting_shred.clone(), shred.payload().clone())
+                    .is_err()
+                {
+                    warn!("bad duplicate store..");
```

Comment:
this was old code but nonetheless we need to log the actual `err` than just dropping it.